rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -288,222 +258,146 @@ public String defineNodeSource(final String sessionId, final String nodeSourceNa
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.infrastructures(sessionId);
-            }
-        });
+    @Override
+    public String getInfrastructures(String sessionId) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.infrastructures(sessionId));
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see org.ow2.proactive_grid_cloud_portal.rm.client.RMService#getPolicies(java.lang.String)
-     */
-    public String getPolicies(final String sessionId) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.policies(sessionId);
-            }
-        });
+    @Override
+    public String getPolicies(String sessionId) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.policies(sessionId));
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see org.ow2.proactive_grid_cloud_portal.rm.client.RMService#lockNodes(java.lang.String,
-     * java.util.Set)
-     */
-    public String lockNodes(final String sessionId, Set<String> urls) throws RestServerException, ServiceException {
-        return executeFunction(new BiFunction<RestClient, Set<String>, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient, Set<String> strings) {
-                return restClient.lockNodes(sessionId, strings);
-            }
-        }, urls, "lock");
+    @Override
+    public String getNodeSourceConfiguration(String sessionId, String nodeSourceName)
+            throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.getNodeSourceConfiguration(sessionId,
+                                                                                                       nodeSourceName));
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see org.ow2.proactive_grid_cloud_portal.rm.client.RMService#unlockNodes(java.lang.String,
-     * java.util.Set)
-     */
-    public String unlockNodes(final String sessionId, Set<String> urls) throws RestServerException, ServiceException {
-        return executeFunction(new BiFunction<RestClient, Set<String>, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient, Set<String> strings) {
-                return restClient.unlockNodes(sessionId, strings);
-            }
-        }, urls, "unlock");
+    @Override
+    public String lockNodes(String sessionId, Set<String> urls) throws RestServerException, ServiceException {
+        return executeFunction((restClient, strings) -> restClient.lockNodes(sessionId, strings), urls, "lock");
     }
 
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)